app/models/application_record.rb

```diff
@@ -3,4 +3,15 @@
 # Base ApplicationRecord Class
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+  validate :druid_is_valid
+
+  # ensure we have a valid druid with prefix
+  def druid_is_valid
+    errors.add(:druid, 'is not valid') unless valid_druid?
+  end
+
+  # @return [boolean]
+  def valid_druid?
+    DruidTools::Druid.valid?(druid, true) && druid.starts_with?('druid:')
+  end
 end
```

Comment:
we only have two db models and both require druid validation, so move here to avoid code duplication (can further abstract if/when we add more models that do not require this validation)

Reply:
We could also consider a lightweight validator class like we use in a handful of other places (H2, ETD, sul_pub, DSA...), like so:

https://github.com/sul-dlss/hydra_etd/blob/main/app/models/submission.rb#L21

https://github.com/sul-dlss/hydra_etd/blob/main/app/validators/folio_instance_hrid_validator.rb

so like:

```ruby
class DruidValidator < ActiveModel::Validator
  def validate(record)
    return if DruidTools::Druid.valid?(record.druid, true) && record.druid.starts_with?('druid:')

    record.errors.add(:druid, 'is not valid')
  end
end
```

Reply:
Sounds good, I'll do this, and add the single validation line explicitly to both models to make it clear